src/Participation/Manager.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace PhpAb\Participation;
+
+use PhpAb\Storage\StorageInterface;
+use PhpAb\Test\TestInterface;
+use PhpAb\Variant\VariantInterface;
+
+class Manager implements ParticipationManagerInterface
+{
+    /**
+     * @var \PhpAb\Storage\StorageInterface
+     */
+    private $storage;
+
+    /**
+     * @var string
+     */
+    private $prefix;
+
+    /**
+     * @param \PhpAb\Storage\StorageInterface $storage The storage which should be used
+     * @param string                          $prefix  The prefix to avoid collision with other stored variables
+     */
+    public function __construct(StorageInterface $storage, $prefix = 'phpab_')
+    {
+        $this->storage = $storage;
+        $this->prefix = $prefix;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function participates($test, $variant = null)
```

Comment:
I don't see the problem at this point. Would leave it as is till we have different use cases.